api/base/permissions.py

```diff
@@ -76,3 +78,19 @@ def has_object_permission(self, request, view, obj):
             obj = self.get_object(request, view, obj)
             return super(Perm, self).has_object_permission(request, view, obj)
     return Perm
+
+class UserIsConfirmed(permissions.BasePermission):
+
+    def has_permission(self, request, view):
+        confirmed = request.user and request.user.is_confirmed
+        if not confirmed:
+            raise UnconfirmedAccountError
+        return True
+
+class UserIsNotDeactivated(permissions.BasePermission):
```

Comment:
This can probably be bundled up with the other class as UserIsActive or similar. It's also possible we could just add this logic into the base user permission class. That being said, the reason we might not is if we expect situations where we need to have the logic be separate. If our API might need to have access to an unconfirmed user but not a deactivated user, for example. I can't think of any reason why we might right now, and we could always break it up again later.